src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -753,6 +759,38 @@ public static ServerHostInfo GetServerHostInfo(IDbConnection connection)
             return hostInfo;
         }
 
+        /// <summary>
+        /// Gets the server host cpu count and memory from sys.dm_os_sys_info view
+        /// </summary>
+        /// <param name="connection">The connection</param>
+        public static ServerSysInfo GetServerSysInfo(IDbConnection connection)
+        {
+            var sysInfo = new ServerSysInfo();
+            try
+            {
+               SqlConnection conn = null;
+               ReliableSqlConnection reliableSqlConnection = connection as ReliableSqlConnection;
+               SqlConnection sqlConnection = connection as SqlConnection;
+                if (reliableSqlConnection != null)
+                {
+                    conn = reliableSqlConnection.GetUnderlyingConnection();
+                }
+                else if (sqlConnection != null)
+                {
+                    conn = sqlConnection;
+                }
+                var server = new Server(new ServerConnection(conn));
+                server.SetDefaultInitFields(server.GetType(), new String[] { "Processor", "PhysicalMemory" });
+                sysInfo.CpuCount = server.Processors;
+                sysInfo.PhysicalMemoryInKb = server.PhysicalMemory;
+            }
+            catch (Exception ex)
+            {
+                Logger.Write(TraceEventType.Error, ex.ToString());
```

Comment:
shouldn't you put a value in the structure to tell the client you couldn't get the right data? The user experience on an error is just going to be seeing zeroes, correct?

Reply:
Typically we would want to throw an exception so that the caller will get an error on their side - I agree that silently just sending 0's is not preferable.